autoload/merginal/buffers/historyLog.vim

```diff
@@ -12,8 +12,10 @@ function! s:f.generateHeader() dict abort
                 \ '']
 endfunction
 
+let g:merginal_log_commit_count = get(g:, 'merginal_log_commit_count', -20)
+
 function! s:f.generateBody() dict abort
-    let l:logLines = self.gitLines('log', '--format=%h %aN%n%ci%n%s%n', self.branch, '--')
+    let l:logLines = self.gitLines('log', g:merginal_log_commit_count,'--format=%h %aN%n%ai%n%s%n', self.branch)
```

Comment:
So - overall I think the line setting it globally should be removed, and this line should look like this:

```vim
let l:logLines = self.gitLines('log', '--max-count', get(g:, 'merginal_logCommitCount, -1), '--format=%h %aN%n%ci%n%s%n', self.branch)
```